game/cards/dm06/guardian.go

```diff
@@ -0,0 +1,54 @@
+package dm06
+
+import (
+	"duel-masters/game/civ"
+	"duel-masters/game/cnd"
+	"duel-masters/game/family"
+	"duel-masters/game/fx"
+	"duel-masters/game/match"
+)
+
+func LuGilaSilverRiftGuardian(c *match.Card) {
+
+	c.Name = "Lu Gila, Silver Rift Guardian"
+	c.Power = 4000
+	c.Civ = civ.Light
+	c.Family = []string{family.Guardian}
+	c.ManaCost = 5
+	c.ManaRequirement = []string{civ.Light}
+
+	c.Use(fx.Creature, fx.Blocker, fx.CantAttackPlayers, func(card *match.Card, ctx *match.Context) {
+
+		if card.Zone != match.BATTLEZONE && card.Zone != match.HIDDENZONE {
+			return
+		}
+
+		if event, ok := ctx.Event.(*match.CardMoved); ok {
+			if event.To != match.BATTLEZONE {
+				return
+			}
+
+			playedCard, ok := ctx.Match.Player1.Player.GetCard(event.CardID, match.BATTLEZONE)
+			if ok != nil {
+				playedCard, ok = ctx.Match.Player2.Player.GetCard(event.CardID, match.BATTLEZONE)
+				if ok != nil {
+					return
+				}
+			}
+
+			if !playedCard.HasCondition(cnd.Evolution) {
+				return
+			}
+
+			// If the card is hidden, make sure it was the one that just evolved
+			at := playedCard.Attachments()
+			if card.Zone == match.HIDDENZONE && at[len(at)-1] != card {
+				return
+			}
```

Comment:
What is this for?

If it is in fact needed then we should probably add a check to make sure we don't panic the program by indexing into `at[-1]` if in the future there's an evolution card that can be played without any attachments for some reason

Reply:
According to the rulings (https://duelmasters.fandom.com/wiki/Lu_Gila,_Silver_Rift_Guardian/Rulings) if you evolve Lu Gila, it's evolution also becomes tapped. But good point with the indexing will add a check for that and improve the explanation.